### PR TITLE
Us7 edit temperature page

### PR DESCRIPTION
**Pull request by TuncAlt**
[US7](https://github.com/TuncAlt/capstone-project/issues/15)

### CODE REVIEW

File: pages/logTempForm.js [hunk 1]
```diff
@@ -2,8 +2,8 @@ import LogTemperatureForm from "@/components/Forms/LogTempForm/LogTempForm";
 
 export default function logTempForm() {
   return (
-    <main>
+    <div>
       <LogTemperatureForm />
-    </main>
+    </div>
```
Comment (stefankorder) [inline pages/logTempForm.js]: Do you really need a `<div>` for this case? Hint: Your `<div>`is surrounding just one element.

File: pages/index.js [hunk 2]
```diff
@@ -2,8 +2,8 @@ import DeviceList from "@/components/DeviceList/deviceList";
 
 export default function Home({ devices }) {
   return (
-    <main>
+    <div>
       <DeviceList devices={devices} />;
-    </main>
+    </div>
```
Comment (stefankorder) [inline pages/index.js]: Do you really need a `<div>` for this case? Hint: Your `<div>`is surrounding just one element.

File: pages/devices/index.js [hunk 3]
```diff
@@ -2,8 +2,8 @@ import DeviceList from "@/components/DeviceList/deviceList";
 
 export default function deviceOverview({ devices }) {
   return (
-    <main>
+    <>
       <DeviceList devices={devices} />;
-    </main>
+    </>
```
Comment (stefankorder) [inline pages/devices/index.js]: Do you really need a `<>` for this case? Hint: Your `<>`is surrounding just one element.

File: components/Navigation/HeaderNavigation.js [hunk 4]
```diff
@@ -43,13 +43,13 @@ export default function HeaderNavigation() {
...
+        isactive={router.pathname === "/addDeviceForm"}
       >
         Add Device
       </StyledAddDeviceLink>
       <StyledLogTempLink
         href="/logTempForm"
-        isActive={router.pathname === "/logTempForm"}
+        isactive={router.pathname === "/logTempForm"}
       >
         {" "}
```
Comment (stefankorder) [inline components/Navigation/HeaderNavigation.js]: Try to avoid empty spaces like this.

File: components/DeviceList/deviceList.js [hunk 5]
```diff
@@ -102,58 +116,66 @@ const StyledHeader = styled.h1`
...
-                  <StyledTempReading>
-                    {device.readings[device.readings.length - 1].temperature}°C
-                  </StyledTempReading>
+                device?.readings[device?.readings?.length - 1]?.temperature >
+                  0 ? (
+                  <>
+                    {Number(device?.readings?.temperature) >
+                    Number(device.maxTemp) ? (
+                      <StyledIconWrapper>
+                        {" "}
```
Comment (stefankorder) [inline components/DeviceList/deviceList.js]: Try to avoid empty spaces like this.

File: components/DeviceList/deviceList.js [hunk 6]
```diff
@@ -102,58 +116,66 @@ const StyledHeader = styled.h1`
...
+                        <FaExclamationTriangle />
+                      </StyledIconWrapper>
+                    ) : Number(device?.readings?.temperature) <
+                      Number(device.minTemp) ? (
+                      <StyledIconWrapper>
+                        {" "}
```
Comment (stefankorder) [inline components/DeviceList/deviceList.js]: Try to avoid empty spaces like this.

File: components/DeviceList/deviceList.js [hunk 7]
```diff
@@ -102,58 +116,66 @@ const StyledHeader = styled.h1`
...
-            Click to add a new device
-          </StyledAddLink>
-        )}
-      </StyledWrapper>
+              </StyledLinkContainer>
+            ))
+          ) : (
+            <StyledAddLink href="/addDeviceForm">
+              Hello there!
+              <br />
```
Comment (stefankorder) [inline components/DeviceList/deviceList.js]: Try to avoid `<br />` try to create space with styling.

File: pages/temperatureReadings.js [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+    return null;
+  }
+
+  // filter the readings based on deviceId
+  const readings = device?.readings?.filter((reading) =>
+    reading.date.toLowerCase().includes(searchTerm.toLowerCase())
+  );
+
+  const handleDeleteReading = (deviceId, readingId) => {
+    console.log("handleDelete:", deviceId, readingId);
```
Comment (stefankorder) [inline pages/temperatureReadings.js]: Delete `console.log()`'s if they are only for debugging.

**Review by stefankorder — CHANGES_REQUESTED** [review]
Looks good so far! Please take a look at my comments.

File: components/DeviceList/deviceList.js (hunk 7 above)
Comment (TuncAlt) [reply components/DeviceList/deviceList.js]: I will do this later on

**Comment by TuncAlt** [comment]
Hi Stefan, thank you for your recommendations. I've resolved it now.

**Comment by TuncAlt** [comment]
> Looks good so far! Please take a look at my comments.

Hi Stefan, thank you for your recommendations. I've resolved it now.

**Review by stefankorder — APPROVED** [review]
👍🏻

**Pull request merged**